fiducia/h5loader.py

```
@@ -0,0 +1,565 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Apr  3 15:42:39 2025
+
+Module for importing newest h5 formatted Dante shots
+
+@author: Daniel H. Barnak
+"""
+
+# python modules
+import h5py
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import re
+
+def h5_import(filepath):
+    """
+    imports h5 file and checks for the proper formatting by reading data keys
+    and attributes. Searches for attributes will be hard-coded to this module,
+    so this is the first check to see if there are errors associated with it.
+
+    Parameters
+    ----------
+    filepath : str or Path object
+        Location and name of the H5 Dante file.
+
+    Returns
+    -------
+    data : h5py object
+        The H5 file heirarchy and datasets.
+
+    """
+    dataImp= h5py.File(filepath)
+    try:
+        # get detector positions
+        keys = np.array(list(dataImp.keys()), dtype = 'str') 
+        attSearch = dataImp[keys[0]][keys[0] + " Baseline"]
+        # get all attributes of first position
+        attrs = np.array(list(attSearch.attrs.keys()))
+    except:
+        print("Data keys and attributes location may have changed! Check file"+
```

How do you want to handle this failure? Should the program stop here or can it continue even with this failure?

If it needs to fail then use `raise Exception` instead of `print`.
If the program can continue, then this would be a good case for using the [warnings module](https://docs.python.org/3/library/warnings.html#module-warnings)

---

I want it to stop here. If there is a file format change we can cross that bridge when we get there. I just wanted something to detect it first.